### PR TITLE
Fixed Net 8: Page and control Unloaded events firing on iOS when navigating to another page

### LINKED ISSUE #21916: Net 8: Page and control Unloaded events firing on iOS when navigating to another page
### Description

Since updating our app to Net 8, when navigating to a new page on iOS, the Unloaded and Loaded events for the starting page are firing, even though that page is still in the stack and is not actually unloaded. In fact, it's actually firing twice. The sequence goes: 
Navigate -> Unloaded -> Loaded -> Unloaded

This doesn't happen on Android and never happened on either platform in Net 7. 

This is causing quite a big headache for us now because many of our pages have event and other subscriptions that get cleaned up in Unloaded when the page is removed from the stack / closed. As a result we now have event handlers and other subs being unregistered incorrectly, leaving the page in a broken / non-functional state when the user navigates back to it

I ran into a similar issue [in this ticket](https://github.com/dotnet/maui/issues/21331#issuecomment-2011404286)

### Steps to Reproduce

if using the included repro project: 

1. Run the UnloadedNet8 app on iOS and put breakpoints in the loaded and unloaded event handlers on MainPage
2. Navigate away and back using the buttons
3. You will see when you click the button on Main Page that Unloaded fires, then Loaded, then Unloaded, and then the page navigates

This does not happen on Android, and does not happen in the UnloadedNet7 app project at all on either platform

### Link to public reproduction project repository

https://gooseza.visualstudio.com/MauiBugs/_git/UnloadedEventBug

### Version with bug

8.0.20 SR4

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

7.0.101

### Affected platforms

iOS

### Affected platform versions

iOS 17.3.1. Not sure if others are affected

### Did you find any workaround?

Not one that I am happy with 

I can possibly try re-registering my event handlers and subscriptions in Loaded or OnAppearing but this is not ideal as it will mean recreating them multiple times instead of just once when the page is created. 

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by NirmalKumarYuvaraj**
### Issue Details:
On the iOS platform, the Loaded and Unloaded events of the current page are triggered twice when navigating to another page.

### Root Cause:
The MovedToWindow method in the MauiView class is called twice when page navigation occurs with animation enabled. This behavior causes the _movedToWindow event to be repeatedly wired and unwired, which sequentially triggers the Loaded and Unloaded events.
 
### Description of Change:
To resolve the issue, I moved the UnloadedCheck and OnLoadedCheck methods to the UI thread. This change ensures that calls are queued on the UI thread, allowing the Loaded and Unloaded events to trigger correctly once the UIView is attached to a window. Additionally, I removed redundant code for wiring the Loaded and Unloaded events, as the MovedToWindow method will handle these triggers for all MauiView instances upon attachment to and detachment from the window, with the events firing directly from _movedToWindow.

**Tested the behaviour in the following platforms.**
- [x] Android
- [x] Windows
- [x] iOS
- [x] Mac
### Reference
https://stackoverflow.com/questions/34484518/willmovetowindow-is-called-twice
### Issues Fixed
Fixes #21916
### Screenshots
**Before Issue Fix**

https://github.com/user-attachments/assets/694ce1cb-fbfc-458c-bd66-0b2dafb26594

**After Issue Fix**

https://github.com/user-attachments/assets/a2878184-c90b-4698-b527-3367723e88fc

### CODE REVIEW

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+		
+		[Test]
+		[Category(UITestCategories.Shell)]
+		public void Shell_Issue21916()
+		{
+			App.WaitForElement("ButtonId");
+			App.Click("ButtonId");
+			App.WaitForElement("ButtonId");
+			App.Click("ButtonId");
+			VerifyScreenshot();
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs:24]: Pending snapshots available in the latest build.
image
Could you commit the images?

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs (hunk 1 above)
Comment (NirmalKumarYuvaraj) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs:24]: @jsuarezruiz , I have added the pending snapshots.

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#if !ANDROID && !MACCATALYST
```
Comment (jsuarezruiz) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs:1]: Why avoid Android? Could be nice to test the same behavior in all the platforms.

Comment (NirmalKumarYuvaraj) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs:1]: On the Android platform, the Unloaded event of the current page is not triggered when navigating to the next page. As a result, the UI test case for Android has been skipped. Let me know if you have any further concerns. 

Android issue link : https://github.com/dotnet/maui/issues/26050

**Review by PureWeen — CHANGES_REQUESTED** [review]
https://github.com/dotnet/maui/pull/25852#discussion_r1853702872

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs (hunk 2 above)
Comment (NirmalKumarYuvaraj) [reply src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue21916.cs:1]: @jsuarezruiz , here is the link to the issue - https://github.com/dotnet/maui/issues/26050

**Review by PureWeen — APPROVED** [review]

**Comment by PureWeen** [comment]
I tried a few different ways to fix this. I'm never a fan of fixing something by just invoke back to the main thread but I'm not really sure how else to resolve this currently. 

One thought I had was that if we just stopped firing unloaded when the view is still part of the xplat tree that would be one way where the number of `MovedToWindows` wouldn't matter. 

i created an issue here for .NET 10 that we can look at
https://github.com/dotnet/maui/issues/26659

which might generally make loaded/unloaded operate more consistently between platforms

**Comment by PureWeen** [comment]
- DelayedIsRefreshingAndCommandTest_SwipeDown is an unrelated failure

**Pull request merged**